packages/headless/src/ssr-next.index.ts

```diff
@@ -36,11 +36,17 @@
  * export type SearchStaticState = InferStaticState<typeof engineDefinition>;
  * export type SearchHydratedState = InferHydratedState<typeof engineDefinition>;
  *
- * export const {
- *   fetchStaticState,
- *   hydrateStaticState,
- *   setNavigatorContextProvider,
- * } = engineDefinition;
+ * export const {fetchStaticState, hydrateStaticState} = engineDefinition;
+
+ * // Pass navigator context directly per call (server-side)
+ * await fetchStaticState({
+ *   navigatorContext: {
+ *     forwardedFor: req.ip,
+ *     referrer: req.headers.referer || null,
+ *     userAgent: req.headers['user-agent'] || null,
+ *     clientId: req.sessionID || 'anonymous',
```

Comment:
This `clientId` is very crucial for a proper implementation, and we should give a proper code sample if that's something we find valuable to do here.

A clientId should be a stable uuidv4, saved in cookie, valid for one year, and reused when the cookie already exists.

If does not have to be full proper sample, but should be verbose enough to guide users properly.

This comment applies to all the instances in this PR where we provide such type of comment/sample.